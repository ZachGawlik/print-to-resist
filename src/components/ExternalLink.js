import React, { PropTypes } from 'react';

const ExternalLink = ({ url }) => (
  <a
    href={url}
    target="_blank"
    rel="noopener noreferrer"
  >
    {url}
  </a>
);

ExternalLink.propTypes = {
  url: PropTypes.string.isRequired,
};

export default ExternalLink;
