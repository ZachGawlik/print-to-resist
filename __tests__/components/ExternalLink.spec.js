import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ExternalLink from '../../src/components/ExternalLink';

describe('ExternalLink', () => {
  it('renders as expected', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<ExternalLink url="google.com" />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
