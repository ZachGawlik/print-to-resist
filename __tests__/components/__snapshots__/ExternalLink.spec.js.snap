// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink renders as expected 1`] = `
<a
  href="google.com"
  rel="noopener noreferrer"
  target="_blank"
>
  google.com
</a>
`;
